ably-ios/ARTAuthTokenParams.m

```diff
@@ -143,7 +145,7 @@ - (ARTAuthTokenRequest *)sign:(NSString *)key {
     NSString *keyName = keyComponents[0];
     NSString *keySecret = keyComponents[1];
     NSString *nonce = generateNonce();
-    NSString *clientId = self.clientId ? self.clientId : @"";
+    NSString *clientId = self.clientId ? self.clientId : @"*";
```

Comment:
> @paddybyers Correct?

No, I don't think so. A `TokenRequest` only contains the wildcard `clientId` if explicitly requested by the caller.

If the caller does not specify a `clientId`, and the Ably library was not initialised with a `clientId`, then the token request does not contain any `clientId`. For the purposes of canonicalisation, if there is no `clientId` then an empty string is used in place of the `clientId` to construct the canonical text.